src/util-inl.h

```diff
@@ -200,11 +224,74 @@ TypeName* Unwrap(v8::Local<v8::Object> object) {
   return static_cast<TypeName*>(pointer);
 }
 
-void SwapBytes(uint16_t* dst, const uint16_t* src, size_t buflen) {
-  for (size_t i = 0; i < buflen; i += 1)
-    dst[i] = (src[i] << 8) | (src[i] >> 8);
+#define SWAP(a, b)   \
+  tmp = data[a];     \
+  data[a] = data[b]; \
+  data[b] = tmp;
+
+void SwapBytes16(char* data, size_t nbytes) {
+  CHECK_EQ(nbytes % 2, 0);
+
+  int align = reinterpret_cast<uintptr_t>(data) % sizeof(uint16_t);
+
+  if (align == 0) {
+    uint16_t* data16 = reinterpret_cast<uint16_t*>(data);
```

Comment:
This is a strict-aliasing violation, strictly speaking.  My PR operates on just char for that reason.